---
title: "Lamb Jaw Length - Heritability estimate and GWAS"
author: "Caelinn James"
output:
 html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
**Written for R `r getRversion()`, last updated `r format(Sys.time(), '%d/%m/%y')`**  
<br/>

### Load packages
```{r, message = FALSE, warning = FALSE}
library(RODBC)
library(data.table)
library(stringr)
library(splitstackshape)
library(qqman)
```
<br/>

### Connect to the database

Note that from database front end version 2.22, it is important to click "Make R Friendly" in the queries section before running the analysis, or loading the pedigree will not work (will not work at all for prior front end versions)
```{r}
dv <- "C:/Users/s1944058/Documents/Sheep/Database/SoaySheep2.30.accdb"
con <- odbcConnectAccess2007(dv)
tbls <- sqlTables(con)
sheep <- sqlFetch(con,"sys_SheepPlusBirthInfo")
Jaw <- sqlFetch(con,"tblLegJawMeasures")
MuseSumm <- sqlFetch(con,"usys_qMuseumSummary")
```


Check to see if the files have loaded correctly (don't worry about negative IDs or lots of NAs)... 
```{r}
head(sheep)
```

```{r}
head(Jaw)
```

```{r}
head(MuseSumm)
```
<br/> 

### Filter the data
First, we need to match jaw records to sheep IDs
```{r}
Jaw <- Jaw[-which(Jaw$Comment %like% "DUPLICATE"),]
Jaw <- merge(Jaw,MuseSumm[,1:17],by="StorageRef",all.x=TRUE,all.y=FALSE)
Jaw <- Jaw[,c("ID","Jaw","Leg","AgeatDeath_months")]
```

And only keep sheep we have jaw lengths for
```{r}
Jaw <- Jaw[!(is.na(Jaw$Jaw)),]
```

Merge the Jaw file with the sheep file
```{r}
Jaw <- merge(Jaw,sheep,by="ID")
```

```{r}
Jaw <- Jaw[!(is.na(Jaw$BirthYear)),]
Jaw <- Jaw[!(is.na(Jaw$Sex)),]
Jaw <- Jaw[!(is.na(Jaw$MumID)),]
Jaw <- Jaw[which(Jaw$AgeatDeath_months < 14),]
Jaw <- Jaw[which(Jaw$AgeatDeath_months > 2),]
```

```{r}
Jaw$ID <- as.factor(Jaw$ID)
Jaw$Sex <- as.factor(Jaw$Sex)
Jaw <- Jaw[which(Jaw$SibCount < 2),]

```

```{r}
SNPs_50K <- read.table("~/Documents/Sheep/50KSNP/Plates_1to87_QC3.fam")
Jaw <- Jaw[which(Jaw$ID %in% SNPs_50K$V2),]
SNPs_HD <- read.table("~/Documents/Sheep/HDSNP/HD_QC.fam")
Jaw <- Jaw[which(Jaw$ID %in% SNPs_HD$V2),]

```

This leaves us with __`r nrow(Jaw)`__ individuals, and the following distribution of jaw lengths:
```{r}
hist(Jaw$Jaw, xlab="Jaw length",main="Histogram of jaw lengths of Soay sheep lambs")

```
<br/>
We will now save a list of phenotypes, fixed effects and random effects
```{r}
JawLength_Lamb_LambPhenotypes <- cbind(rep(1,nrow(Jaw)),as.character(Jaw$ID),as.character(Jaw$Jaw))

JawLength_Lamb_LambAssoc <- cbind(rep(1,nrow(Jaw)),as.character(Jaw$ID),Jaw$Sex,Jaw$SibCount)

JawLength_Lamb_LambQAssoc <- cbind(rep(1,nrow(Jaw)),as.character(Jaw$ID),Jaw$AgeatDeath_months)

JawLength_Lamb_LambRandom <-cbind(rep(1,nrow(Jaw)),as.character(Jaw$ID),Jaw$BirthYear,Jaw$MumID)
colnames(JawLength_Lamb_LambRandom) <- c("FID","IID","BirthYear","MumID")


write.table(JawLength_Lamb_LambPhenotypes,file="~/Documents/Sheep/R/H2_GWAS/JawLength/JawLength_LambPhenotypes.txt",row.names = F,col.names = F,quote = F)

write.table(JawLength_Lamb_LambAssoc,file="~/Documents/Sheep/R/H2_GWAS/JawLength/JawLength_LambAssoc.txt",row.names = F,col.names = F,quote = F)

write.table(JawLength_Lamb_LambQAssoc,file="~/Documents/Sheep/R/H2_GWAS/JawLength/JawLength_LambQAssoc.txt",row.names = F,col.names = F,quote = F)

write.table(JawLength_Lamb_LambRandom,file="~/Documents/Sheep/R/H2_GWAS/JawLength/JawLength_LambRandom.txt",row.names = F,col.names = T,quote = F)
```


## Heritability Estimates  
### 50K  
>#!/bin/sh    
>#\$\ -cwd  
>#\$\ -l h_rt=1:00:00  
>#\$\ -pe sharedmem 1  
>#\$\ -l h_vmem=100G  
>#\$\ -M s1944058@sms.ed.ac.uk  
>#\$\ -m baes    
>. /etc/profile.d/modules.sh  
>  
>module load igmm/apps/dissect/1.15.2c  
>module load igmm/apps/plink/1.90b4  
>  
>mpirun -np 8 dissect.mpich  \-\-reml \-\-bfile ../InputData/Plates_1to87_QC3  \-\-pheno JawLengthPhenotypes.txt \-\-pheno-col 1 \-\-blue \-\-covar JawLengthAssoc.txt \-\-qcovar JawLengthQAssoc.txt \-\-random-effects JawLengthRandom.txt \-\-random-effects-cols 1 2 \-\-out 50K/JawLength_Lamb_50K_H2  
>  
>if [[ -f 50K/JawLength_Lamb_50K_H2.badsnps ]]  
>then    
>plink \-\-bfile ../InputData/Plates_1to87_QC3 \-\-sheep \-\-exclude 50K/JawLength_Lamb_50K_H2.badsnps \-\-make-bed \-\-out 50K/JawLength_Lamb_50K_H2  
>
>mpirun -np 8 dissect.mpich  \-\-reml \-\-bfile 50K/JawLength_Lamb_50K_H2 \-\-pheno JawLengthPhenotypes.txt \-\-pheno-col 1 \-\-blue \-\-covar JawLengthAssoc.txt \-\-qcovar JawLengthQAssoc.txt \-\-random-effects JawLengthRandom.txt \-\-random-effects-cols 1 2 \-\-out 50K/JawLength_Lamb_50K_H2  
>fi 
> 
>mpirun -np 8 dissect.mpich \-\-reml \-\-bfile ../InputData/Plates_1to87_QC3 \-\-pheno JawLengthPhenotypes.txt \-\-pheno-col 1 \-\-blue \-\-random-effects JawLengthRandom.txt \-\-random-effects-cols 1 2 \-\-out 50K/JawLength_Lamb_50K_TotalPhenotypeVariance  
>  
>if [[ -f 50K/JawLength_Lamb_50K_TotalPhenotypeVariance.badsnps ]]   
>then   
>plink \-\-bfile ../InputData/Plates_1to87_QC3 \-\-sheep \-\-exclude 50K/JawLength_Lamb_50K_TotalPhenotypeVariance.badsnps \-\-make-bed \-\-out 50K/JawLength_Lamb_50K_TotalPhenotypeVariance  
>  
>mpirun -np 8 dissect.mpich  \-\-reml \-\-bfile 50K/JawLength_Lamb_50K_TotalPhenotypeVariance \-\-pheno JawLengthPhenotypes.txt \-\-pheno-col 1 \-\-blue \-\-random-effects JawLengthRandom.txt \-\-random-effects-cols 1 2 \-\-out 50K/JawLength_Lamb_50K_TotalPhenotypeVariance  
>fi  

```{r,include=FALSE}
Assoc <- read.table("C:/Users/s1944058/Documents/Sheep/R/H2_GWAS/JawLength/Lamb/50K/JawLength_Lamb_50K_H2.blue.discrete",header=TRUE)
Assoc$NAME <- c("Sex","Twins","Triplets")
QAssoc <- read.table("C:/Users/s1944058/Documents/Sheep/R/H2_GWAS/JawLength/Lamb/50K/JawLength_Lamb_50K_H2.blue.quantitative",header=TRUE)
QAssoc$NAME <- c("Age at death")
FixedEffects_50K <- rbind(Assoc,QAssoc)

Random <- read.csv("C:/Users/s1944058/Documents/Sheep/R/H2_GWAS/JawLength/Lamb/50K/JawLength_Lamb_50K_H2.reml")
RandomEffects_50K <- cSplit(Random,"X...Summary.results.",sep= " ")
RandomEffects_50K <- RandomEffects_50K[c(2:5,8),1:3]
colnames(RandomEffects_50K) <- colnames(FixedEffects_50K)
```
```{r}
FixedEffects_50K
RandomEffects_50K
```

### HD  
>#!/bin/sh    
>#\$\ -cwd  
>#\$\ -l h_rt=1:00:00  
>#\$\ -pe sharedmem 1  
>#\$\ -l h_vmem=100G  
>#\$\ -M s1944058@sms.ed.ac.uk  
>#\$\ -m baes    
>. /etc/profile.d/modules.sh    
>  
>module load igmm/apps/dissect/1.15.2c  
>module load igmm/apps/plink/1.90b4  
>  
>mpirun -np 8 dissect.mpich  \-\-reml \-\-bfile ../InputData/HD_QC \-\-pheno JawLengthPhenotypes.txt \-\-pheno-col 1 \-\-blue \-\-covar JawLengthAssoc.txt \-\-qcovar JawLengthQAssoc.txt \-\-random-effects JawLengthRandom.txt \-\-random-effects-cols 1 2 \-\-out HD/JawLength_Lamb_HD_H2  
>  
>if [[ -f HD/JawLength_Lamb_HD_H2.badsnps ]]  
>then    
>plink \-\-bfile ../InputData/HD_QC \-\-sheep \-\-exclude HD/JawLength_Lamb_HD_H2.badsnps \-\-make-bed \-\-out HD/JawLength_Lamb_HD_H2  
>
>mpirun -np 8 dissect.mpich  \-\-reml \-\-bfile HD/JawLength_Lamb_HD_H2 \-\-pheno JawLengthPhenotypes.txt \-\-pheno-col 1 \-\-blue \-\-covar JawLengthAssoc.txt \-\-qcovar JawLengthQAssoc.txt \-\-random-effects JawLengthRandom.txt \-\-random-effects-cols 1 2 \-\-out HD/JawLength_Lamb_HD_H2  
>fi 
>  
>mpirun -np 8 dissect.mpich  \-\-reml \-\-bfile ../InputData/HD_QC \-\-pheno JawLengthPhenotypes.txt \-\-pheno-col 1 \-\-blue \-\-random-effects JawLengthRandom.txt \-\-random-effects-cols 1 2 \-\-out HD/JawLength_Lamb_HD_TotalPhenotypeVariance  
>  
>if [[ -f HD/JawLength_Lamb_HD_TotalPhenotypeVariance.badsnps ]]   
>then    
>plink \-\-bfile ../InputData/HD_QC \-\-sheep \-\-exclude HD/JawLength_Lamb_HD_TotalPhenotypeVariance.badsnps \-\-make-bed \-\-out HD/JawLength_Lamb_HD_TotalPhenotypeVariance  
>  
>mpirun -np 8 dissect.mpich  \-\-reml \-\-bfile HD/JawLength_Lamb_HD_TotalPhenotypeVariance \-\-pheno JawLengthPhenotypes.txt \-\-pheno-col 1 \-\-blue \-\-random-effects JawLengthRandom.txt \-\-random-effects-cols 1 2 \-\-out HD/JawLength_Lamb_HD_TotalPhenotypeVariance  
>fi  

```{r,include=FALSE}
Assoc <- read.table("C:/Users/s1944058/Documents/Sheep/R/H2_GWAS/JawLength/Lamb/HD/JawLength_Lamb_HD_H2.blue.discrete",header=TRUE)
Assoc$NAME <- c("Sex","Twins","Triplets")
QAssoc <- read.table("C:/Users/s1944058/Documents/Sheep/R/H2_GWAS/JawLength/Lamb/HD/JawLength_Lamb_HD_H2.blue.quantitative",header=TRUE)
QAssoc$NAME <- c("Age at death")
FixedEffects_HD <- rbind(Assoc,QAssoc)

Random <- read.csv("C:/Users/s1944058/Documents/Sheep/R/H2_GWAS/JawLength/Lamb/HD/JawLength_Lamb_HD_H2.reml")
RandomEffects_HD <- cSplit(Random,"X...Summary.results.",sep= " ")
RandomEffects_HD <- RandomEffects_HD[c(2:5,8),1:3]
colnames(RandomEffects_HD) <- colnames(FixedEffects_HD)
```
```{r}
FixedEffects_HD
RandomEffects_HD
```

### GWAS  
>#!/bin/sh  
>#\$\ -cwd  
>#\$\ -l h_rt=2:00:00  
>#\$\ -pe sharedmem 1  
>#\$\ -l h_vmem=200G  
>#\$\ -t 1-26  
>#\$\ -tc 5  
>#\$\ -M s1944058@sms.ed.ac.uk  
>#\$\ -m baes   
>. /etc/profile.d/modules.sh  
>
>module load igmm/apps/dissect/1.15.2c  
>module load igmm/apps/plink/1.90b4  
>  
>i=\$\SGE_TASK_ID  
>  
>plink \-\-bfile ../InputData/Plates_1to87_QC3 \-\-sheep \-\-not-chr \$\{i},X \-\-keep JawLengthPhenotypes.txt \-\-make-bed \-\-out 50K/NoChr\$\{i}  
>  
>mpirun -np 8 dissect.mpich  \-\-make-grm \-\-bfile 50K/NoChr\$\{i} \-\-out 50K/NoChr\$\{i}_GRM  
>  
>echo "50K/Chr\$\{i} 50K/NoChr\$\{i}_GRM" > 50K/GWAS_GRM\$\{i}.txt  
>  
>plink \-\-bfile ../InputData/Plates_1to87_QC3 \-\-sheep \-\-chr \$\{i} \-\-keep JawLengthPhenotypes.txt \-\-make-bed \-\-out 50K/Chr\$\{i}  
>  
>if [[ -f 50K/NoChr\$\{i}_GRM.badsnps ]]  
>then  
>plink \-\-bfile 50K/NoChr\$\{i} \-\-sheep \-\-exclude 50K/NoChr\$\{i}_GRM.badsnps  \-\-make-bed \-\-out 50K/NoChr\$\{i}_2  
>  
>mpirun -np 8 dissect.mpich  \-\-make-grm \-\-bfile 50K/NoChr\$\{i}_2 \-\-out 50K/NoChr\$\{i}_GRM
>  
>fi  
>  
>mpirun -np 8 dissect.mpich  \-\-gwas \-\-bfile-grm-list 50K/GWAS_GRM\$\{i}.txt \-\-pheno JawLengthPhenotypes.txt \-\-covar JawLengthAssoc.txt \-\-qcovar JawLengthQAssoc.txt \-\-random-effects JawLengthRandom.txt \-\-random-effects-cols 1,2  \-\-out 50K/JawLength_Lamb_50K_GWAS_Chr\$\{i}  
>  
>if [[ -f 50K/JawLength_Lamb_50K_GWAS_Chr\$\{i}.badsnps ]]  
>then  
>plink \-\-bfile 50K/Chr\$\{i} \-\-sheep \-\-exclude 50K/JawLength_Lamb_50K_GWAS_Chr\$\{i}.badsnps \-\-make-bed \-\-out 50K/Chr\$\{i}_2  
>  
>echo "50K/Chr\$\{i}_2 50K/NoChr\$\{i}_GRM" > 50K/GWAS_GRM\$\{i}.txt  
>  
> mpirun -np 8 dissect.mpich  \-\-gwas \-\-bfile-grm-list 50K/GWAS_GRM\$\{i}.txt \-\-pheno JawLengthPhenotypes.txt \-\-covar JawLengthAssoc.txt \-\-qcovar JawLengthQAssoc.txt \-\-random-effects JawLengthRandom.txt \-\-random-effects-cols 1,2 \-\-out 50K/JawLength_Lamb_50K_GWAS_Chr\$\{i}    
> fi 

```{r,fig.width=14}
files = list.files(pattern="JawLength_Lamb_50K_GWAS_Chr",recursive=TRUE)

SNPresults = do.call(rbind, lapply(files, function(x) read.table(x,header=T)))

bimFile <- read.table("C:/Users/s1944058/Documents/Sheep/50KSNP/Plates_1to87_QC3.bim",header=F)

GWAS50KResults <- merge(bimFile,SNPresults[,c("SNP","PV"),],by.x="V2",by.y="SNP",all.x=FALSE,all.y=TRUE)
colnames(GWAS50KResults) <- c("SNP","Chr","V3","Pos","Ref","Minor","PV")
GWAS50KResults$Chr <- as.numeric(GWAS50KResults$Chr)
par(mfrow=c(1,2))
manhattan(GWAS50KResults,chr="Chr",bp="Pos",p="PV",snp="SNP", main = "Lamb jaw length Manhattan plot (50K)", genomewideline = -log10(0.05/20082),suggestiveline = FALSE)
qq(GWAS50KResults$PV,main="Lamb jaw length QQ plot (50K)")
```
```{r} 
SNPs <- GWAS50KResults[which(GWAS50KResults$PV < (0.05/20082) ),c("Chr","Pos","SNP","PV")]
SNPs[order(SNPs$PV),]
```
<br/>

### Imputed SNP data

>#!/bin/sh  
>#\$\ -cwd  
>#\$\ -l h_rt=3:00:00  
>#\$\ -pe sharedmem 1  
>#\$\ -l h_vmem=200G  
>#\$\ -t 1-26  
>#\$\ -tc 5  
>#\$\ -M s1944058@sms.ed.ac.uk  
>#\$\ -m baes   
>. /etc/profile.d/modules.sh  
>
>module load igmm/apps/dissect/1.15.2c  
>module load igmm/apps/plink/1.90b4  
>  
>i=\$\SGE_TASK_ID  
>  
>plink \-\-bfile ../InputData/HD_QC \-\-sheep \-\-not-chr \$\{i},X \-\-keep JawLengthPhenotypes.txt \-\-make-bed \-\-out HD/NoChr\$\{i}  
>  
>mpirun -np 8 dissect.mpich  \-\-make-grm \-\-bfile HD/NoChr\$\{i} \-\-out HD/NoChr\$\{i}_GRM  
>  
>echo "HD/Chr\$\{i} HD/NoChr\$\{i}_GRM" > HD/GWAS_GRM\$\{i}.txt  
>  
>plink \-\-bfile ../InputData/HD_QC \-\-sheep \-\-chr \$\{i} \-\-keep JawLengthPhenotypes.txt \-\-make-bed \-\-out HD/Chr\$\{i}  
>  
>if [[ -f HD/NoChr\$\{i}_GRM.badsnps ]]  
>then  
>plink \-\-bfile HD/NoChr\$\{i} \-\-sheep \-\-exclude HD/NoChr\$\{i}_GRM.badsnps  \-\-make-bed \-\-out HD/NoChr\$\{i}_2  
>  
>mpirun -np 8 dissect.mpich  \-\-make-grm \-\-bfile HD/NoChr\$\{i}_2 \-\-out HD/NoChr\$\{i}_GRM  
>  
>fi  
>  
>mpirun -np 8 dissect.mpich  \-\-gwas \-\-bfile-grm-list HD/GWAS_GRM\$\{i}.txt \-\-pheno JawLengthPhenotypes.txt \-\-covar JawLengthAssoc.txt \-\-qcovar JawLengthQAssoc.txt \-\-random-effects JawLengthRandom.txt \-\-random-effects-cols 1,2  \-\-out HD/JawLength_Lamb_HD_GWAS_Chr\$\{i}  
>  
>if [[ -f HD/JawLength_Lamb_HD_GWAS_Chr\$\{i}.badsnps ]]  
>then  
>plink \-\-bfile HD/Chr\$\{i} \-\-sheep \-\-exclude HD/JawLength_Lamb_HD_GWAS_Chr\$\{i}.badsnps \-\-make-bed \-\-out HD/Chr\$\{i}_2  
>
>echo "HD/Chr\$\{i}_2 HD/NoChr\$\{i}_GRM" > HD/GWAS_GRM\$\{i}.txt  
>  
>mpirun -np 8 dissect.mpich  \-\-gwas \-\-bfile-grm-list HD/GWAS_GRM\$\{i}.txt \-\-pheno JawLengthPhenotypes.txt \-\-covar JawLengthAssoc.txt \-\-qcovar JawLengthQAssoc.txt \-\-random-effects JawLengthRandom.txt \-\-random-effects-cols 1,2  \-\-out HD/JawLength_Lamb_HD_GWAS_Chr\$\{i}    
>fi 
 

```{r,fig.width=14}
files = list.files(pattern="JawLength_Lamb_HD_GWAS_Chr",recursive=TRUE)

SNPresults = do.call(rbind, lapply(files, function(x) read.table(x,header=T)))

bimFile <- read.table("C:/Users/s1944058/Documents/Sheep/HDSNP/HD_QC.bim",header=F)

GWASHDResults <- merge(bimFile,SNPresults[,c("SNP","PV"),],by.x="V2",by.y="SNP",all.x=FALSE,all.y=TRUE)
colnames(GWASHDResults) <- c("SNP","Chr","V3","Pos","Ref","Minor","PV")
GWASHDResults$Chr <- as.numeric(GWASHDResults$Chr)
par(mfrow=c(1,2))
manhattan(GWASHDResults,chr="Chr",bp="Pos",p="PV",snp="SNP", main = "Lamb jaw length Manhattan plot (HD)", genomewideline = -log10(0.05/48635),suggestiveline = FALSE)
qq(GWASHDResults$PV,main="Lamb jaw length QQ plot (HD)")
```
```{r} 
SNPs <- GWASHDResults[which(GWASHDResults$PV < (0.05/48635) ),c("Chr","Pos","SNP","PV")]
SNPs[order(SNPs$PV),]
```
<br/>

## **FINAL RESULTS**
### 50K
```{r,fig.width=14}
RandomEffects_50K[nrow(RandomEffects_50K),]
par(mfrow=c(1,2))
manhattan(GWAS50KResults,chr="Chr",bp="Pos",p="PV",snp="SNP", main = "Lamb jaw length Manhattan plot (50K)", genomewideline = -log10(0.05/20082),suggestiveline = FALSE)
qq(GWAS50KResults$PV,main="Lamb jaw length QQ plot (50K)")
```

### HD    
```{r,fig.width=14}
RandomEffects_HD[nrow(RandomEffects_HD),]
par(mfrow=c(1,2))
manhattan(GWASHDResults,chr="Chr",bp="Pos",p="PV",snp="SNP", main = "Lamb jaw length Manhattan plot (HD)", genomewideline = -log10(0.05/48635),suggestiveline = FALSE)
qq(GWASHDResults$PV,main="Lamb jaw length QQ plot (HD)")
```
