---
title: "Adult hindleg length - Heritability estimate and GWAS"
author: "Caelinn James"
output:
 html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
**Written for R `r getRversion()`, last updated `r format(Sys.time(), '%d/%m/%y')`**  

<br/>

### Load packages
```{r, message = FALSE, warning = FALSE}
library(RODBC)
library(data.table)
library(stringr)
library(splitstackshape)
library(qqman)
```
<br/>

### Connect to the database

Note that from database front end version 2.22, it is important to click "Make R Friendly" in the queries section behind running the analysis, or loading the pedigree will not work (will not work at all for prior front end versions)

```{r}
dv <- "C:/Users/s1944058/Documents/Sheep/Database/SoaySheep2.30.accdb"
con <- odbcConnectAccess2007(dv)
tbls <- sqlTables(con)
capturedata <- sqlFetch(con,"CaptureData")
sheep <- sqlFetch(con,"sys_SheepPlusBirthInfo")
```
Check to see if the files have loaded correctly (don't worry about negative IDs or lots of NAs)...  
First the capture data file....
```{r}
head(capturedata)
```
Then the sheep file...
```{r}
head(sheep)
```
<br/>

### Filter the data
We are only interested in looking at capture events that happened in August and to adults
```{r}
capturedataadult <- merge(sheep,capturedata,by.x="ID")
capturedataadult<- capturedataadult[which(capturedataadult$CapMonth == 8),]
capturedataadult <- capturedataadult[which(capturedataadult$CapYear > (capturedataadult$BirthYear +1)),]
```

Now, only include records of capture events that measured the hindleg.
```{r}
capturedataadult <- capturedataadult[!is.na(capturedataadult$Hindleg),]
```

Calculate the age of the sheep for their hindleg measure
```{r}
capturedataadult$Age <- capturedataadult$CapYear - capturedataadult$BirthYear
capturedataadult$Age <- as.numeric(capturedataadult$Age)
```

Find the population density the year of the measurement
```{r}
Villdens <- sqlFetch(con,"HirtaVillPop")
capturedataadult <- merge(capturedataadult,Villdens[,c("VillTotal","Year")],by.x="BirthYear",by.y="Year",all.x=T)
capturedataadult <- capturedataadult[!is.na(capturedataadult$VillTotal),]
```


To ensure the same individuals are used for all three heritability estimates, remove any sheep that are not included in the genotype data
```{r}
SNPs_50K <- read.table("~/Documents/Sheep/50KSNP/Plates_1to87_QC3.fam")
capturedataadult <- capturedataadult[which(capturedataadult$ID %in% SNPs_50K$V2),]
SNPs_HD <- read.table("~/Documents/Sheep/HDSNP/HD_QC.fam")
capturedataadult <- capturedataadult[which(capturedataadult$ID %in% SNPs_HD$V2),]

```
<br/>

This leaves us with __`r nrow(capturedataadult)`__ records from __`r length(unique(capturedataadult$ID))`__ individuals, and the following distribution of leg lengths:
```{r}
hist(as.numeric(capturedataadult$Hindleg),xlab="Hindleg length (mm)",main="Histogram of hindleg lengths in Soay sheep adults")
```
<br/>

We will now save a list of phenotypes, fixed effects and random effects.
```{r}
AdulthindLegPhenotypes <- cbind(rep(1,nrow(capturedataadult)),as.numeric(as.character(capturedataadult$CapRef)),as.character(capturedataadult$ID),as.character(capturedataadult$Hindleg))

AdulthindLegAssoc <- cbind(rep(1,nrow(capturedataadult)),as.numeric(as.character(capturedataadult$CapRef)),capturedataadult$Sex)

AdulthindLegQAssoc <-cbind(rep(1,nrow(capturedataadult)),as.numeric(as.character(capturedataadult$CapRef)),capturedataadult$VillTotal,capturedataadult$Age)

AdulthindLegRandom <-cbind(rep(1,nrow(capturedataadult)),as.numeric(as.character(capturedataadult$CapRef)),capturedataadult$CapYear,as.character(capturedataadult$ID))
colnames(AdulthindLegRandom) <- c("FID","IID","CapYear","ID")


write.table(AdulthindLegPhenotypes,file="~/Documents/Sheep/R/H2_GWAS/Hindleg/AdultHindLegPhenotypes.txt",row.names = F,col.names = F,quote = F)

write.table(AdulthindLegAssoc,file="~/Documents/Sheep/R/H2_GWAS/Hindleg/AdultHindLegAssoc.txt",row.names = F,col.names = F,quote = F)

write.table(AdulthindLegQAssoc,file="~/Documents/Sheep/R/H2_GWAS/Hindleg/AdultHindLegQAssoc.txt",row.names = F,col.names = F,quote = F)

write.table(AdulthindLegRandom,file="~/Documents/Sheep/R/H2_GWAS/Hindleg/AdultHindLegRandom.txt",row.names = F,col.names = T,quote = F)
```
<br/>

## Prep the genotype files  
>#!/bin/sh  
>#\$\ -cwd  
>#\$\ -l h_rt=0:30:00  
>#\$\ -pe sharedmem 1  
>#\$\ -l h_vmem=50G
>  
>sort -k 3,3 AdultHindlegPhenotypes.txt > Pheno150K.txt  
>awk '($2 >0)' ../InputData/Plates_1to87_QC3.ped > Ped150K.ped  
>sort -k 2,2 Ped150K.ped > Ped250K.ped  
>join -1 3 -2 2 Pheno150K.txt Ped250K.ped > Ped350K.ped  
>cut -d ' ' -f 1,4,5 \-\-complement Ped350K.ped > AdultHindleg50K.ped  
>  
>sort -k 3,3 AdultHindlegPhenotypes.txt > Pheno1HD.txt  
>awk '($2 >0)' ../InputData/HD_QC.ped > Ped1HD.ped  
>sort -k 2,2 Ped1HD.ped > Ped2HD.ped  
>join -1 3 -2 2 Pheno1HD.txt Ped2HD.ped > Ped3HD.ped  
>cut -d ' ' -f 1,4,5 \-\-complement Ped3HD.ped > AdultHindlegHD.ped  

## Heritability Estimates  
### 50K
>#!/bin/sh    
>#\$\ -cwd  
>#\$\ -l h_rt=1:00:00  
>#\$\ -pe sharedmem 1  
>#\$\ -l h_vmem=100G  
>#\$\ -M s1944058@sms.ed.ac.uk  
>#\$\ -m baes    
>. /etc/profile.d/modules.sh  
>  
>module load igmm/apps/dissect/1.15.2c  
>module load igmm/apps/plink/1.90b4  
>  
>plink --file ../InputData/Plates_1to87_QC3 \-\-ped AdultHindleg50K.ped \-\-sheep \-\-not-chr 0,X \-\-make-bed \-\-out 50K/AdultHindleg50K  
>  
>mpirun -np 8 dissect.mpich  \-\-reml \-\-bfile 50K/AdultHindleg50K  \-\-pheno AdultHindlegPhenotypes.txt \-\-pheno-col 2 \-\-blue \-\-covar AdultHindlegAssoc.txt \-\-qcovar AdultHindlegQAssoc.txt \-\-random-effects AdultHindlegRandom.txt \-\-random-effects-cols 1 2 \-\-out 50K/AdultHindLeg_50K_H2  
>  
>if [[ -f 50K/AdultHindLeg_50K_H2.badsnps ]]  
>then    
>plink \-\-bfile ../InputData/Plates_1to87_QC3 \-\-sheep \-\-exclude 50K/AdultHindLeg_50K_H2.badsnps \-\-make-bed \-\-out 50K/AdultHindLeg_50K_H2  
>
>mpirun -np 8 dissect.mpich  \-\-reml \-\-bfile 50K/AdultHindLeg_50K_H2 \-\-pheno AdultHindlegPhenotypes.txt \-\-pheno-col 2 \-\-blue \-\-covar AdultHindlegAssoc.txt \-\-qcovar AdultHindlegQAssoc.txt \-\-random-effects AdultHindlegRandom.txt \-\-random-effects-cols 1 2 \-\-out 50K/AdultHindLeg_50K_H2  
>fi 
> 
>mpirun -np 8 dissect.mpich \-\-reml \-\-bfile 50K/AdultHindleg50K \-\-pheno AdultHindlegPhenotypes.txt \-\-pheno-col 2 \-\-blue \-\-random-effects AdultHindlegRandom.txt \-\-random-effects-cols 1 2 \-\-out 50K/AdultHindLeg_50K_TotalPhenotypeVariance  
>  
>if [[ -f 50K/AdultHindLeg_50K_TotalPhenotypeVariance.badsnps ]]   
>then   
>plink \-\-bfile ../InputData/Plates_1to87_QC3 \-\-sheep \-\-exclude 50K/AdultHindLeg_50K_TotalPhenotypeVariance.badsnps \-\-make-bed \-\-out 50K/AdultHindLeg_50K_TotalPhenotypeVariance  
>  
>mpirun -np 8 dissect.mpich  \-\-reml \-\-bfile 50K/AdultHindLeg_50K_TotalPhenotypeVariance \-\-pheno AdultHindlegPhenotypes.txt \-\-pheno-col 2 \-\-blue \-\-random-effects AdultHindlegRandom.txt \-\-random-effects-cols 1 2 \-\-out 50K/AdultHindLeg_50K_TotalPhenotypeVariance  
>fi  

```{r,include=FALSE}
Assoc <- read.table("C:/Users/s1944058/Documents/Sheep/R/H2_GWAS/Hindleg/Adult/50K/AdultHindLeg_50K_H2.blue.discrete",header=TRUE)
Assoc$NAME <- c("Sex")
QAssoc <- read.table("C:/Users/s1944058/Documents/Sheep/R/H2_GWAS/Hindleg/Adult/50K/AdultHindLeg_50K_H2.blue.quantitative",header=TRUE)
QAssoc$NAME <- c("Population","Age")
FixedEffects_50K <- rbind(Assoc,QAssoc)

Random <- read.csv("C:/Users/s1944058/Documents/Sheep/R/H2_GWAS/Hindleg/Adult/50K/AdultHindLeg_50K_H2.reml")
RandomEffects_50K <- cSplit(Random,"X...Summary.results.",sep= " ")
RandomEffects_50K <- RandomEffects_50K[c(2:5,8),1:3]
colnames(RandomEffects_50K) <- colnames(FixedEffects_50K)
```
```{r}
FixedEffects_50K
RandomEffects_50K
```

### HD
>#!/bin/sh    
>#\$\ -cwd  
>#\$\ -l h_rt=1:00:00  
>#\$\ -pe sharedmem 1  
>#\$\ -l h_vmem=100G  
>#\$\ -M s1944058@sms.ed.ac.uk  
>#\$\ -m baes    
>. /etc/profile.d/modules.sh  
>  
>module load igmm/apps/dissect/1.15.2c  
>module load igmm/apps/plink/1.90b4  
>  
>plink --file ../InputData/HD_QC \-\-ped AdultHindlegHD.ped \-\-sheep \-\-not-chr 0,X \-\-make-bed \-\-out HD/AdultHindlegHD  
>  
>mpirun -np 8 dissect.mpich  \-\-reml \-\-bfile HD/AdultHindlegHD  \-\-pheno AdultHindlegPhenotypes.txt \-\-pheno-col 2 \-\-blue \-\-covar AdultHindlegAssoc.txt \-\-qcovar AdultHindlegQAssoc.txt \-\-random-effects AdultHindlegRandom.txt \-\-random-effects-cols 1 2 \-\-out HD/AdultHindLeg_HD_H2  
>  
>if [[ -f HD/AdultHindLeg_HD_H2.badsnps ]]  
>then    
>plink \-\-bfile ../InputData/Plates_1to87_QC3 \-\-sheep \-\-exclude HD/AdultHindLeg_HD_H2.badsnps \-\-make-bed \-\-out HD/AdultHindLeg_HD_H2  
>
>mpirun -np 8 dissect.mpich  \-\-reml \-\-bfile HD/AdultHindLeg_HD_H2 \-\-pheno AdultHindlegPhenotypes.txt \-\-pheno-col 2 \-\-blue \-\-covar AdultHindlegAssoc.txt \-\-qcovar AdultHindlegQAssoc.txt \-\-random-effects AdultHindlegRandom.txt \-\-random-effects-cols 1 2 \-\-out HD/AdultHindLeg_HD_H2  
>fi 
> 
>mpirun -np 8 dissect.mpich \-\-reml \-\-bfile HD/AdultHindlegHD \-\-pheno AdultHindlegPhenotypes.txt \-\-pheno-col 2 \-\-blue \-\-random-effects AdultHindlegRandom.txt \-\-random-effects-cols 1 2 \-\-out HD/AdultHindLeg_HD_TotalPhenotypeVariance  
>  
>if [[ -f HD/AdultHindLeg_HD_TotalPhenotypeVariance.badsnps ]]   
>then   
>plink \-\-bfile ../InputData/Plates_1to87_QC3 \-\-sheep \-\-exclude HD/AdultHindLeg_HD_TotalPhenotypeVariance.badsnps \-\-make-bed \-\-out HD/AdultHindLeg_HD_TotalPhenotypeVariance  
>  
>mpirun -np 8 dissect.mpich  \-\-reml \-\-bfile HD/AdultHindLeg_HD_TotalPhenotypeVariance \-\-pheno AdultHindlegPhenotypes.txt \-\-pheno-col 2 \-\-blue \-\-random-effects AdultHindlegRandom.txt \-\-random-effects-cols 1 2 \-\-out HD/AdultHindLeg_HD_TotalPhenotypeVariance  
>fi  

```{r,include=FALSE}
Assoc <- read.table("C:/Users/s1944058/Documents/Sheep/R/H2_GWAS/Hindleg/Adult/HD/AdultHindLeg_HD_H2.blue.discrete",header=TRUE)
Assoc$NAME <- c("Sex")
QAssoc <- read.table("C:/Users/s1944058/Documents/Sheep/R/H2_GWAS/Hindleg/Adult/HD/AdultHindLeg_HD_H2.blue.quantitative",header=TRUE)
QAssoc$NAME <- c("Population","Age")
FixedEffects_HD <- rbind(Assoc,QAssoc)

Random <- read.csv("C:/Users/s1944058/Documents/Sheep/R/H2_GWAS/Hindleg/Adult/HD/AdultHindLeg_HD_H2.reml")
RandomEffects_HD <- cSplit(Random,"X...Summary.results.",sep= " ")
RandomEffects_HD <- RandomEffects_HD[c(2:5,8),1:3]
colnames(RandomEffects_HD) <- colnames(FixedEffects_HD)
```
```{r}
FixedEffects_HD
RandomEffects_HD
```
### GWAS  
>#!/bin/sh  
>#\$\ -cwd  
>#\$\ -l h_rt=2:00:00  
>#\$\ -pe sharedmem 1  
>#\$\ -l h_vmem=200G  
>#\$\ -t 1-26  
>#\$\ -tc 5  
>#\$\ -M s1944058@sms.ed.ac.uk  
>#\$\ -m baes   
>. /etc/profile.d/modules.sh  
>
>module load igmm/apps/dissect/1.15.2c  
>module load igmm/apps/plink/1.90b4  
>  
>i=\$\SGE_TASK_ID  
>  
>plink \-\-bfile ../InputData/Plates_1to87_QC3 \-\-ped AdultHindleg50K.ped \-\-sheep \-\-not-chr \$\{i},X \-\-keep AdultHindlegPhenotypes.txt \-\-make-bed \-\-out 50K/NoChr\$\{i}  
>  
>mpirun -np 8 dissect.mpich  \-\-make-grm \-\-bfile 50K/NoChr\$\{i} \-\-out 50K/NoChr\$\{i}_GRM  
>  
>echo "50K/Chr\$\{i} 50K/NoChr\$\{i}_GRM" > 50K/GWAS_GRM\$\{i}.txt  
>  
>plink \-\-bfile ../InputData/Plates_1to87_QC3 \-\-ped AdultHindleg50K.ped \-\-sheep \-\-chr \$\{i} \-\-keep AdultHindlegPhenotypes.txt \-\-make-bed \-\-out 50K/Chr\$\{i}  
>  
>if [[ -f 50K/NoChr\$\{i}_GRM.badsnps ]]  
>then  
>plink \-\-bfile 50K/NoChr\$\{i} \-\-sheep \-\-exclude 50K/NoChr\$\{i}_GRM.badsnps  \-\-make-bed \-\-out 50K/NoChr\$\{i}_2  
>  
>mpirun -np 8 dissect.mpich  \-\-make-grm \-\-bfile 50K/NoChr\$\{i}_2 \-\-out 50K/NoChr\$\{i}_GRM
>  
>fi  
>  
>mpirun -np 8 dissect.mpich  \-\-gwas \-\-bfile-grm-list 50K/GWAS_GRM\$\{i}.txt \-\-pheno AdultHindlegPhenotypes.txt \-\-pheno-col 2 \-\-covar AdultHindlegAssoc.txt \-\-qcovar AdultHindlegQAssoc.txt \-\-random-effects AdultHindlegRandom.txt \-\-random-effects-cols 1 2  \-\-out 50K/AdultHindLeg_50K_GWAS_Chr\$\{i}  
>  
>if [[ -f 50K/AdultHindLeg_50K_GWAS_Chr\$\{i}.badsnps ]]  
>then  
>plink \-\-bfile 50K/Chr\$\{i} \-\-sheep \-\-exclude 50K/AdultHindLeg_50K_GWAS_Chr\$\{i}.badsnps \-\-make-bed \-\-out 50K/Chr\$\{i}_2  
>  
>echo "50K/Chr\$\{i}_2 50K/NoChr\$\{i}_GRM" > 50K/GWAS_GRM\$\{i}.txt  
>  
> mpirun -np 8 dissect.mpich  \-\-gwas \-\-bfile-grm-list 50K/GWAS_GRM\$\{i}.txt \-\-pheno AdultHindlegPhenotypes.txt \-\-pheno-col 2 \-\-covar AdultHindlegAssoc.txt \-\-qcovar AdultHindlegQAssoc.txt \-\-random-effects AdultHindlegRandom.txt \-\-random-effects-cols 1 2 \-\-out 50K/AdultHindLeg_50K_GWAS_Chr\$\{i}    
> fi 

```{r,fig.width=14}
files = list.files(pattern="AdultHindLeg_50K_GWAS_Chr",recursive=TRUE)

SNPresults = do.call(rbind, lapply(files, function(x) read.table(x,header=T)))

bimFile <- read.table("C:/Users/s1944058/Documents/Sheep/50KSNP/Plates_1to87_QC3.bim",header=F)

GWAS50KResults <- merge(bimFile,SNPresults[,c("SNP","PV"),],by.x="V2",by.y="SNP",all.x=FALSE,all.y=TRUE)
colnames(GWAS50KResults) <- c("SNP","Chr","V3","Pos","Ref","Minor","PV")
GWAS50KResults$Chr <- as.numeric(GWAS50KResults$Chr)
par(mfrow=c(1,2))
manhattan(GWAS50KResults,chr="Chr",bp="Pos",p="PV",snp="SNP", main = "Adult hindleg length Manhattan plot (50K)", genomewideline = -log10(0.05/20082),suggestiveline = FALSE)
qq(GWAS50KResults$PV,main="Adult hindleg length QQ plot (50K)")
```
```{r} 
SNPs <- GWAS50KResults[which(GWAS50KResults$PV < (0.05/20082) ),c("Chr","Pos","SNP","PV")]
SNPs[order(SNPs$PV),]
table(SNPs$Chr)
```
<br/>

### Imputed SNP data

>#!/bin/sh  
>#\$\ -cwd  
>#\$\ -l h_rt=3:00:00  
>#\$\ -pe sharedmem 1  
>#\$\ -l h_vmem=200G  
>#\$\ -t 1-26  
>#\$\ -tc 5  
>#\$\ -M s1944058@sms.ed.ac.uk  
>#\$\ -m baes   
>. /etc/profile.d/modules.sh  
>
>module load igmm/apps/dissect/1.15.2c  
>module load igmm/apps/plink/1.90b4  
>  
>i=\$\SGE_TASK_ID  
>  
>plink \-\-bfile ../InputData/HD_QC \-\-ped AdultHindlegHD.ped \-\-sheep \-\-not-chr \$\{i},X \-\-keep AdultHindlegPhenotypes.txt \-\-make-bed \-\-out HD/NoChr\$\{i}  
>  
>mpirun -np 8 dissect.mpich  \-\-make-grm \-\-bfile HD/NoChr\$\{i} \-\-out HD/NoChr\$\{i}_GRM  
>  
>echo "HD/Chr\$\{i} HD/NoChr\$\{i}_GRM" > HD/GWAS_GRM\$\{i}.txt  
>  
>plink \-\-bfile ../InputData/HD_QC \-\-ped AdultHindlegHD.ped \-\-sheep \-\-chr \$\{i} \-\-keep AdultHindlegPhenotypes.txt \-\-make-bed \-\-out HD/Chr\$\{i}  
>  
>if [[ -f HD/NoChr\$\{i}_GRM.badsnps ]]  
>then  
>plink \-\-bfile HD/NoChr\$\{i} \-\-sheep \-\-exclude HD/NoChr\$\{i}_GRM.badsnps  \-\-make-bed \-\-out HD/NoChr\$\{i}_2  
>  
>mpirun -np 8 dissect.mpich  \-\-make-grm \-\-bfile HD/NoChr\$\{i}_2 \-\-out HD/NoChr\$\{i}_GRM  
>  
>fi  
>  
>mpirun -np 8 dissect.mpich  \-\-gwas \-\-bfile-grm-list HD/GWAS_GRM\$\{i}.txt \-\-pheno AdultHindlegPhenotypes.txt \-\-pheno-col 2 \-\-covar AdultHindlegAssoc.txt \-\-qcovar AdultHindlegQAssoc.txt \-\-random-effects AdultHindlegRandom.txt \-\-random-effects-cols 1 2  \-\-out HD/AdultHindLeg_HD_GWAS_Chr\$\{i}  
>  
>if [[ -f HD/AdultHindLeg_HD_GWAS_Chr\$\{i}.badsnps ]]  
>then  
>plink \-\-bfile HD/Chr\$\{i} \-\-sheep \-\-exclude HD/AdultHindLeg_HD_GWAS_Chr\$\{i}.badsnps \-\-make-bed \-\-out HD/Chr\$\{i}_2  
>
>echo "HD/Chr\$\{i}_2 HD/NoChr\$\{i}_GRM" > HD/GWAS_GRM\$\{i}.txt  
>  
>mpirun -np 8 dissect.mpich  \-\-gwas \-\-bfile-grm-list HD/GWAS_GRM\$\{i}.txt \-\-pheno AdultHindlegPhenotypes.txt \-\-pheno-col 2 \-\-covar AdultHindlegAssoc.txt \-\-qcovar AdultHindlegQAssoc.txt \-\-random-effects AdultHindlegRandom.txt \-\-random-effects-cols 1 2  \-\-out HD/AdultHindLeg_HD_GWAS_Chr\$\{i}    
>fi 
 

```{r,fig.width=14}
files = list.files(pattern="AdultHindLeg_HD_GWAS_Chr",recursive=TRUE)

SNPresults = do.call(rbind, lapply(files, function(x) read.table(x,header=T)))

bimFile <- read.table("C:/Users/s1944058/Documents/Sheep/HDSNP/HD_QC.bim",header=F)

GWASHDResults <- merge(bimFile,SNPresults[,c("SNP","PV"),],by.x="V2",by.y="SNP",all.x=FALSE,all.y=TRUE)
colnames(GWASHDResults) <- c("SNP","Chr","V3","Pos","Ref","Minor","PV")
GWASHDResults$Chr <- as.numeric(GWASHDResults$Chr)
par(mfrow=c(1,2))
manhattan(GWASHDResults,chr="Chr",bp="Pos",p="PV",snp="SNP", main = "Adult hindleg length Manhattan plot (HD)", genomewideline = -log10(0.05/48635),suggestiveline = FALSE)
qq(GWASHDResults$PV,main="Adult hindleg length QQ plot (HD)")
```
```{r} 
SNPs <- GWASHDResults[which(GWASHDResults$PV < (0.05/48635) ),c("Chr","Pos","SNP","PV")]
SNPs[order(SNPs$PV),]
table(SNPs$Chr)
```
<br/>

## **FINAL RESULTS**
### 50K
```{r,fig.width=14}
RandomEffects_50K[nrow(RandomEffects_50K),]
par(mfrow=c(1,2))
manhattan(GWAS50KResults,chr="Chr",bp="Pos",p="PV",snp="SNP", main = "Adult hindleg length Manhattan plot (50K)", genomewideline = -log10(0.05/20082),suggestiveline = FALSE)
qq(GWAS50KResults$PV,main="Adult hindleg length QQ plot (50K)")
```

### HD    
```{r,fig.width=14}
RandomEffects_HD[nrow(RandomEffects_HD),]
par(mfrow=c(1,2))
manhattan(GWASHDResults,chr="Chr",bp="Pos",p="PV",snp="SNP", main = "Adult hindleg length Manhattan plot (HD)", genomewideline = -log10(0.05/48635),suggestiveline = FALSE)
qq(GWASHDResults$PV,main="Adult hindleg length QQ plot (HD)")
```


```{r, include=FALSE}
png(file="C:/Users/s1944058/Documents/Sheep/R/H2_GWAS/Hindleg/Adult_Hindleg_GWAS.png",width=827,height=583)
par(mfcol=c(2,2),oma=c(0,0,2,0))
par(mar=c(0,5,3,3))
manhattan(GWAS50KResults,chr="Chr",bp="Pos",p="PV",snp="SNP", main = "50K SNP data", genomewideline = -log10(0.05/20082),suggestiveline = FALSE,ylim=c(0,15))
par(mar=c(5,7,3,5))
qq(GWAS50KResults$PV,ylim=c(0,15),xlim=c(0,15))
par(mar=c(0,5,3,3))

manhattan(GWASHDResults,chr="Chr",bp="Pos",p="PV",snp="SNP", main = "Imputed SNP data", genomewideline = -log10(0.05/49781),suggestiveline = FALSE,ylim=c(0,15))
par(mar=c(5,7,3,5))

qq(GWASHDResults$PV,ylim=c(0,15),xlim=c(0,15))
mtext("Adult Hindleg GWAS",outer=TRUE,cex=1.5)

dev.off()
```

```{r, include=FALSE}
png(file="C:/Users/s1944058/Documents/Sheep/R/H2_GWAS/Hindleg/Adult_Hindleg_GWAS.png",width=827,height=583)
par(mfcol=c(2,2),oma=c(0,0,2,0))
par(mar=c(0,5,3,3))
manhattan(GWAS50KResults,chr="Chr",bp="Pos",p="PV",snp="SNP", main = "50K SNP data", genomewideline = -log10(0.05/20082),suggestiveline = FALSE,ylim=c(0,15))
par(mar=c(5,7,3,5))
qq(GWAS50KResults$PV,ylim=c(0,15),xlim=c(0,15))
par(mar=c(0,5,3,3))

manhattan(GWASHDResults,chr="Chr",bp="Pos",p="PV",snp="SNP", main = "Imputed SNP data", genomewideline = -log10(0.05/48635),suggestiveline = FALSE,ylim=c(0,15))
par(mar=c(5,7,3,5))

qq(GWASHDResults$PV,ylim=c(0,15),xlim=c(0,15))
mtext("Adult Hindleg GWAS",outer=TRUE,cex=1.5)

dev.off()
```
